pkg/cvo/sync_worker.go

```diff
@@ -893,6 +893,7 @@ func (w *SyncWorker) apply(ctx context.Context, work *SyncWork, maxWorkers int,
 	}
 	payloadUpdate := w.payload
 
+	w.lock.Lock() // lock to access w.status
```

Comment:
Is the only `status` access in this block `w.status.Architecture`?  Maybe we can:

```go
w.lock.Lock() // lock to access w.status
architecture := w.status.Architecture
w.lock.Unlock()
```

to make it clear that neither `Initializing()` nor `Reconciling()` care about the lock?  Those functions are just enum->bool conversion at the moment, but that wasn't clear (to me anyway) until I popped over to `payload.go` to check, and just because they're quick, isolated checks today doesn't mean they will stay that way forever.

Reply:
Great suggestion, will do!